pkg/manifest/anaconda_installer_iso_tree.go

```diff
@@ -522,47 +541,86 @@ func (p *AnacondaInstallerISOTree) tarPayloadStages() []*osbuild.Stage {
 // Create the base kickstart stage with any options required for unattended
 // installation if set and with any extra file insertion stage required for
 // extra kickstart content.
-func (p *AnacondaInstallerISOTree) makeKickstartStages(kickstartOptions *osbuild.KickstartStageOptions) []*osbuild.Stage {
+func (p *AnacondaInstallerISOTree) makeKickstartStages(stageOptions *osbuild.KickstartStageOptions) []*osbuild.Stage {
+	kickstartOptions := p.Kickstart
+	if kickstartOptions == nil {
+		kickstartOptions = new(kickstart.Options)
+	}
+
 	stages := make([]*osbuild.Stage, 0)
-	if p.UnattendedKickstart {
+
+	// users, groups, and other kickstart options are not allowed when users
+	// add their own kickstarts
+	if kickstartOptions.UserFile != nil {
+		// check if any other option is set and panic - these combinations
```

Comment:
Maybe worthwhile to move the validation policy into `kickstart.go` ? This might be helpful in other places too where we need to validate that the kickstart options work(?). Something like:
```go
diff --git a/pkg/customizations/kickstart/kickstart.go b/pkg/customizations/kickstart/kickstart.go
index 70918e3f9..bf20e700b 100644
--- a/pkg/customizations/kickstart/kickstart.go
+++ b/pkg/customizations/kickstart/kickstart.go
@@ -1,6 +1,10 @@
 package kickstart
 
-import "github.com/osbuild/images/pkg/customizations/users"
+import (
+       "fmt"
+
+       "github.com/osbuild/images/pkg/customizations/users"
+)
 
 type File struct {
        Contents string
@@ -45,3 +49,27 @@ type Options struct {
        // User-defined kickstart files that will be added to the ISO
        UserFile *File
 }
+
+func (kickstartOptions *Options) Validate() error {
+       // users, groups, and other kickstart options are not allowed when users
+       // add their own kickstarts
+       if kickstartOptions.UserFile != nil {
+               // check if any other option is set and panic - these combinations
+               // should be verified by the caller
+               if kickstartOptions.Unattended {
+                       return fmt.Errorf("kickstart unattended options are not compatible with user-supplied kickstart content")
+               }
+
+               if len(kickstartOptions.SudoNopasswd) > 0 {
+                       return fmt.Errorf("kickstart sudo nopasswd %q drop-in file creation is not compatible with user-supplied kickstart content", kickstartOptions.SudoNopasswd)
+               }
+
+               // options are usually already initialised from outside this function
+               // with the payload options (ostree commit or tarball), but might also
+               // have Users and Groups added
+               if len(kickstartOptions.Users)+len(kickstartOptions.Groups) > 0 {
+                       return fmt.Errorf("kickstart users %v and/or groups %v are not compatible with user-supplied kickstart content", kickstartOptions.Users, kickstartOptions.Groups)
+               }
+       }
+       return nil
+}
diff --git a/pkg/manifest/anaconda_installer_iso_tree.go b/pkg/manifest/anaconda_installer_iso_tree.go
index 10816bfcc..51e0d588c 100644
--- a/pkg/manifest/anaconda_installer_iso_tree.go
+++ b/pkg/manifest/anaconda_installer_iso_tree.go
@@ -424,16 +424,9 @@ func (p *AnacondaInstallerISOTree) bootcInstallerKickstartStages() []*osbuild.St
        }
 
        if p.Kickstart.UserFile != nil {
-               // when a user defines their own kickstart, we create a kickstart that
-               // takes care of the installation and let the user kickstart handle
-               // everything else
-
-               // users and groups are NOT allowed when users add their own kickstarts
-               if len(kickstartOptions.Users)+len(kickstartOptions.Groups) > 0 {
-                       // this is a programming error - the combinations should have been verified already
-                       panic("kickstart users and/or groups are not compatible
-               // when a user defines their own kickstart, we create a kickstart that
-               // takes care of the installation and let the user kickstart handle
-               // everything else
-
-               // users and groups are NOT allowed when users add their own kickstarts
-               if len(kickstartOptions.Users)+len(kickstartOptions.Groups) > 0 {
-                       // this is a programming error - the combinations should have been verified already
-                       panic("kickstart users and/or groups are not compatible with user-supplied kickstart content")
+               if err := p.Kickstart.Validate(); err != nil {
+                       panic(err)
                }
-
                stages = append(stages, osbuild.NewKickstartStage(kickstartOptions))
                kickstartFile, err := kickstartOptions.IncludeRaw(p.Kickstart.UserFile.Contents)
                if err != nil {

```